storage/scylla_api.c

```diff
@@ -0,0 +1,616 @@
+  if (src == NULL || size == 0) {
+    return SC_STORAGE_INVAILD_INPUT;
+  }
+
+  *target = malloc(sizeof(cass_byte_t) * (size + 1));
+  if (*target == NULL) {
+    SCYLLA_LOG_ERROR("%s\n", "malloc failure");
+    return SC_STORAGE_OOM;
+  }
+  memcpy(*target, src, size);
+  *((*target) + size) = '\0';
+
+  return SC_OK;
+}
+
+static status_t mallocAndCopyText(char** target, const char* src, size_t size) {
+  if (src == NULL || size == 0) {
+    return SC_STORAGE_INVAILD_INPUT;
+  }
+  *target = malloc(sizeof(char) * (size + 1));
+  if (*target == NULL) {
+    SCYLLA_LOG_ERROR("%s\n", "malloc failure");
+    return SC_STORAGE_OOM;
+  }
+  memcpy(*target, src, size);
+  *((*target) + size) = '\0';
+
+  return SC_OK;
+}
+static status_t get_blob(const CassRow* row, cass_byte_t** target, const char* column) {
+  size_t len;
+  const cass_byte_t* buf;
+  status_t ret = SC_OK;
+  cass_value_get_bytes(cass_row_get_column_by_name(row, column), &buf, &len);
+  ret = mallocAndCopyBlob(target, buf, len);
+  return ret;
+}
+
+static status_t get_text(const CassRow* row, char** target, const char* column) {
+  size_t len;
+  const char* buf;
+  status_t ret = SC_OK;
+  cass_value_get_string(cass_row_get_column_by_name(row, column), &buf, &len);
+  ret = mallocAndCopyText(target, buf, len);
+  return ret;
+}
+
+static status_t create_bundle_table(CassSession* session) {
+  status_t ret = SC_OK;
+
+  if (execute_query(session,
+                    "CREATE KEYSPACE IF NOT EXISTS permanent WITH replication = { \
+                                              'class': 'SimpleStrategy', 'replication_factor': '3' };") != CASS_OK) {
+    ret = SC_STORAGE_CASS_ERROR;
+    goto exit;
+  }
+  if (execute_query(session, "DROP TABLE IF EXISTS permanent.bundleTable;") != CASS_OK) {
+    ret = SC_STORAGE_CASS_ERROR;
+    goto exit;
+  }
+  if (execute_query(session,
+                    "CREATE TABLE IF NOT EXISTS permanent.bundleTable (bundle blob, \
+                                              address blob, \
+                                              transaction blob, \
+                                              message text, \
+                                              value bigint, \
+                                              timestamp bigint, \
+                                              trunk blob, \
+                                              branch blob, \
+                                              PRIMARY KEY (bundle, address, transaction));") != CASS_OK) {
+    ret = SC_STORAGE_CASS_ERROR;
+    goto exit;
+  }
+exit:
+  return ret;
+}
+
+static status_t create_edge_table(CassSession* session) {
+  status_t ret = SC_OK;
+  if (execute_query(session, "DROP TABLE IF EXISTS permanent.edgeTable;") != CASS_OK) {
+    ret = SC_STORAGE_CASS_ERROR;
+    goto exit;
+  }
+  if (execute_query(session,
+                    "CREATE TABLE IF NOT EXISTS permanent.edgeTable (edge blob, \
```

Comment:
Can you avoid multi-line string?

Reply:
Do you mean that do not use \ to 
Instead, modify it as following to prevent random spaces?
![image](https://user-images.githubusercontent.com/3009797/63766321-80852000-c8fd-11e9-8d6f-fb62e40480cc.png)

Reply:
Don't attach images with text.

Reply:
The above change looks good to me.